mlir-sycl/lib/Transforms/Inliner.cpp

```diff
@@ -652,7 +720,25 @@ bool Inliner::inlineCallsInSCC(Inliner &Inliner, CGUseList &UseList,
 
 void Inliner::collectCallOps(CallGraphNode &SrcNode, CallGraph &CG,
                              SymbolTableCollection &SymTable,
-                             SmallVectorImpl<ResolvedCall> &Calls) {
+                             SmallVectorImpl<ResolvedCall> &Calls) const {
+  auto Collect = [this](const CallGraphNode &CGN, const CallOpInterface &Call) {
+    if (CGN.isExternal())
+      return false;
+
+    // Select which call operations to collect based on heuristics.
+    switch (Heuristic.InlineMode) {
+    case sycl::InlineMode::Ludicrous:
+      return isa<sycl::SYCLCallOp>(Call) ||
+             isa<sycl::SYCLConstructorOp>(Call) || isSYCLMethod(Call);
```

Comment:
```suggestion
      return isa<sycl::SYCLCallOp, sycl::SYCLConstructorOp, sycl::SYCLMethodOpInterface>(Call);
```
Same can be applied to all cases.